### DISCUSSION TITLE
LazyExtras markdown; How to configure markdownlint?

### DISCUSSION THREAD

**Discussion by lunchboxer** [LazyVim/LazyVim#2268] [Q&A] [answered]
I have installed the LazyExtras package for markdown. It seems to mostly work as expected without configuration. I would like to globally disable a rule for markdownlint. I'm getting complaints that my lines are more than 80 characters and I don't actually care about that rule. I thought I could accomplish this by creating a file `~/.markdownlint.yaml` with the contents `MD013: false`

This unfortunately had no effect. Can I put something in `~/.config/nvim/lua/plugins/somethingsomething.lua` to tell markdownlint to ignore rule 'MD013'? 

If I put the .markdownlint.yaml in the same directory as the file in the current buffer, then bingo, works as it should. I want to make global configuration though. Having such a file in every directory I use markdown files would get pretty silly, pretty fast. Any ideas what is going on here?

**Comment by dpetka2001** [comment, marked as answer]
Can you try the following
```lua
-- ~/.config/nvim/lua/plugins/lint.lua
return {
  {
    "mfussenegger/nvim-lint",
    opts = {
      linters = {
        markdownlint = {
          args = { "--disable", "MD013", "--" },
        },
      },
    },
  },
}
```
Test if it works how you want.

**Reply by Stierno** [reply]
Not the OP, but thanks. This did the trick.

**Reply by Rash419** [reply]
For some reason it doesn't disable it for me : (

**Reply by dpetka2001** [reply]
It doesn't disable it for you, because this suggested answer is for a different linter than the LazyVim default at the moment. Please take a look [here](https://github.com/LazyVim/LazyVim/discussions/4094#discussioncomment-10178217) that is about `markdownlint-cli2` that LazyVim uses currently.

**Reply by Rash419** [reply]
Thanks, it worked <3

**Comment by wcheek** [comment]
A more local solution that autoformats the lines.

For the `MD013` line length problem check out [this issue](https://github.com/LazyVim/LazyVim/discussions/2512)! 

The configuration is actually having to do with `prettier`, which is doing the actual `markdown` formatting for you.

Make a new file `.prettierrc` local to your `.md` files and add the following configuration:

```json
{
  "printWidth": 80,
  "proseWrap": "always"
}

```

**Reply by rogeriopvl** [reply]
@wcheek that works nicely except for code blocks inside markdown files.

~~I remember seeing somewhere that there's an option in markdownlint that allows to ignore MD013 for code blocks (something like `ignore_code_blocks`), but I can´t find it now and have no idea where to put that in the config.~~

### Update

I found the solution, in case someone is interested as well.

The option to ignore the `line_length` (MD013) rule for code blocks is documented [here](https://github.com/DavidAnson/markdownlint/blob/main/doc/md013.md).

If you want to ignore this rule for code blocks in a given project just create a file named `.markdownlint.jsonc` in the root directory of your project with the content:

```json
{
  "MD013": { "code_blocks": false }
}

```
If you want this config/rule to be applied to all projects, move this file to your home directory and add to lazyvim the following config for **nvim-lint**:

```lua
-- ~/.config/nvim/lua/plugins/lint.lua
return {
  {
    "mfussenegger/nvim-lint",
    opts = {
      linters = {
        markdownlint = {
          args = { "--config", "~/.markdownlint.jsonc", "--" },
        },
      },
    },
  },
}
```
I'm not sure if there's a simpler way to achieve this, but this worked for me.

**Reply by Old-Farmer** [reply]
Very good answer, but I have a question about the answer: what does last "--" work in args?

**Reply by qaxi** [reply]
> Very good answer, but I have a question about the answer: what does last "--" work in args?

in unix cli "--" means: "no more options follow, even if it begins with '-' "

**Reply by curiousignite** [reply]
This does not work for me for some reason.

**Reply by yangxiaobai945** [reply]
> This does not work for me for some reason.

I just added this to ~/.config/nvim/lua/config/autocmds.lua, and it worked for me.There's no need to create ~/.config/nvim/lua/plugins/lint.lua
```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = "markdown",
  callback = function()
    vim.opt_local.spell = false
  end,
})
```

**Reply by richardguinness** [reply]
Thanks for this, I had to adapt slightly for my instance:

- With a `.md` file open, `:lua print(vim.inspect(require('lint').linters_by_ft[vim.bo.filetype]))` to identify which markdown linter has been set up by LazyVim. In my case it's `markdownlint-cli2` (not `markdownlint`).
- Change `markdownlint = {` to `["markdownlint-cli2"] ={` in the snippet above.
- (I changed the name of the referenced `.jsonc` file too)

```lua
return {
  {
    "mfussenegger/nvim-lint",
    opts = {
      linters = {
        ["markdownlint-cli2"] = {
          args = { "--config", "~/.markdownlint-cli2.jsonc", "--" },
        },
      },
    },
  },
}
```
